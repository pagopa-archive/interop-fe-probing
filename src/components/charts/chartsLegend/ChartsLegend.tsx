import { Checkbox, FormControlLabel, Grid } from "@mui/material";

interface ILegendElement {
  label: string;
  color: string;
}

interface IProps {
  legendElements: ILegendElement[];
}

export const ChartsLegend: React.FC<IProps> = ({ legendElements }) => {
  return (
    <Grid container direction="column">
      {legendElements.map((item: ILegendElement) => (
        <Grid item key={item.label}>
          <FormControlLabel
            label={item.label}
            sx={{
              cursor: "default !important",
              "& .MuiTypography-root": {
                color: "#17324D !important",
              },
              "& .MuiButtonBase-root": {
                color: "white !important",
                padding: "0px 9px",
              },
            }}
            disabled
            control={
              <Checkbox
                sx={{
                  "& .MuiSvgIcon-root": {
                    zIndex: 1,
                  },
                  "& .PrivateSwitchBase-input": {
                    width: "auto",
                    height: "auto",
                    top: "auto",
                    left: "auto",
                    opacity: "1",
                    visibility: "hidden",

                    "&::before": {
                      content: '""',
                      position: "absolute",
                      background: `${item.color}`,
                      height: "100%",
                      width: "100%",
                      visibility: "visible",
                    },
                  },
                }}
              />
            }
          />
        </Grid>
      ))}
    </Grid>
  );
};
