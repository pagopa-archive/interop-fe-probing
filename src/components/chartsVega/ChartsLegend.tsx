import { Checkbox, FormControlLabel, Grid } from "@mui/material";

interface ILegendElement {
  label: string;
  color: string;
}

interface IProps {
  legendElements: ILegendElement[];
}

const ChartsLegend = ({ legendElements }: IProps) => {
  return (
    <Grid container direction="column">
      {legendElements.map((item: ILegendElement) => (
        <Grid item key={item.label}>
          <FormControlLabel
            label={item.label}
            sx={{
              cursor: "default !important",
              "& .MuiTypography-root": {
                color: "#17324D !important",
              },
              "& .MuiButtonBase-root": {
                color: "white !important",
                padding: "0px 9px",
              },
            }}
            disabled
            control={
              <Checkbox
                sx={{
                  "& .MuiSvgIcon-root": {
                    zIndex: 1,
                  },
                  "& .PrivateSwitchBase-input": {
                    width: "auto",
                    height: "auto",
                    top: "auto",
                    left: "auto",
                    opacity: "1",
                    visibility: "hidden",

                    "&::before": {
                      content: '""',
                      position: "absolute",
                      background: `${item.color}`,
                      height: "100%",
                      width: "100%",
                      visibility: "visible",
                    },
                  },
                }}
              />
            }
          />
        </Grid>
      ))}
    </Grid>
  );
};
export default ChartsLegend;
