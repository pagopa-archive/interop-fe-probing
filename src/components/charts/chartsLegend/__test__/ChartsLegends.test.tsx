import { describe, expect, test, vi } from "vitest";
import { screen, render } from "@testing-library/react";
import { ChartsLegend } from "../ChartsLegend";

const legendElements = [
  { label: "E-service online", color: "#17324D" },
  { label: "Monitoraggio sospeso", color: "#A2ADB8" },
  { label: "E-service offline", color: "#FE6666" },
];

describe("ChartsLegend", () => {
  test("render component", () => {
    const { container } = render(
      <ChartsLegend legendElements={legendElements} />
    );
    expect(container).toMatchSnapshot();
  });
});
