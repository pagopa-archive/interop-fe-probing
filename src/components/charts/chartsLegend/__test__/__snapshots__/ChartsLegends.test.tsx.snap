// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartsLegend > render component 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <label
        class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-q066m-MuiFormControlLabel-root"
      >
        <span
          aria-disabled="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary css-1cd81rx-MuiButtonBase-root-MuiCheckbox-root"
          tabindex="-1"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            disabled=""
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-ahj2mt-MuiTypography-root"
        >
          E-service online
        </span>
      </label>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <label
        class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-q066m-MuiFormControlLabel-root"
      >
        <span
          aria-disabled="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary css-z4ms8h-MuiButtonBase-root-MuiCheckbox-root"
          tabindex="-1"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            disabled=""
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-ahj2mt-MuiTypography-root"
        >
          Monitoraggio sospeso
        </span>
      </label>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <label
        class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-q066m-MuiFormControlLabel-root"
      >
        <span
          aria-disabled="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary css-7jg725-MuiButtonBase-root-MuiCheckbox-root"
          tabindex="-1"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            disabled=""
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-ahj2mt-MuiTypography-root"
        >
          E-service offline
        </span>
      </label>
    </div>
  </div>
</div>
`;
